zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -1259,7 +1276,11 @@ impl<F: Field> ExecutionConfig<F> {
             ExecutionState::BeginTx => assign_exec_step!(self.begin_tx_gadget),
             ExecutionState::EndTx => assign_exec_step!(self.end_tx_gadget),
             ExecutionState::EndBlock => assign_exec_step!(self.end_block_gadget),
-            ExecutionState::InvalidTx => assign_exec_step!(self.invalid_tx),
+            ExecutionState::InvalidTx => {
+                if let Some(invalid_tx) = &self.invalid_tx {
+                    assign_exec_step!(invalid_tx)
+                }
```

Comment:
Maybe can just unwrap it here and let it error out if it's not set up so it doesn't silently skip it?

Reply:
Nice catch! I fixed it here. [e4be4d0](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1728/commits/e4be4d03736fcd9f80848b4db131181b8b5d1b00)